src/Nethermind/Nethermind.BeaconNode.OApi/BeaconNodeOApiAdapter.cs

```diff
@@ -41,289 +41,339 @@ public class BeaconNodeOApiAdapter : IBeaconNodeOApiController
-                    new Eth1Data(
-                        new Root(beacon_block.Body.Eth1_data.Deposit_root),
-                        (ulong) beacon_block.Body.Eth1_data.Deposit_count,
-                        new Bytes32(beacon_block.Body.Eth1_data.Block_hash)
-                    ),
-                    new Bytes32(beacon_block.Body.Graffiti),
-                    beacon_block.Body.Proposer_slashings.Select(x => new ProposerSlashing(
-                        new ValidatorIndex((ulong) x.Proposer_index),
-                        new SignedBeaconBlockHeader(MapBeaconBlockHeader(x.Header_1), BlsSignature.Zero),
-                        new SignedBeaconBlockHeader(MapBeaconBlockHeader(x.Header_2), BlsSignature.Zero)
-                    )),
-                    beacon_block.Body.Attester_slashings.Select(x => new AttesterSlashing(
-                        MapIndexedAttestation(x.Attestation_1),
-                        MapIndexedAttestation(x.Attestation_2)
-                    )),
-                    beacon_block.Body.Attestations.Select(x =>
-                        new Core2.Containers.Attestation(
-                            new BitArray(x.Aggregation_bits),
-                            MapAttestationData(x.Data),
-                            new BlsSignature(x.Signature)
-                        )
-                    ),
-                    beacon_block.Body.Deposits.Select(x =>
-                        new Core2.Containers.Deposit(
-                            x.Proof.Select(y => new Bytes32(y)),
-                            new DepositData(
-                                new BlsPublicKey(x.Data.Pubkey),
-                                new Bytes32(x.Data.Withdrawal_credentials),
-                                new Gwei((ulong) x.Data.Amount),
-                                new BlsSignature(x.Data.Signature)
-                            )
-                        )
-                    ),
-                    beacon_block.Body.Voluntary_exits.Select(x =>
-                        new Core2.Crypto.SignedVoluntaryExit(
-                        new Core2.Containers.VoluntaryExit(
-                            new Epoch((ulong) x.Epoch),
-                            new ValidatorIndex((ulong) x.Validator_index)
-                            ),
-                            BlsSignature.Zero
-                        )
-                    )
-                )
-            );
-            SignedBeaconBlock signedBlock = new SignedBeaconBlock(block, new BlsSignature(Bytes.FromHexString(beacon_block.Signature)));
-            bool acceptedLocally = await _beaconNode.PublishBlockAsync(signedBlock, CancellationToken.None);
-            
-            // TODO: return 200 or 202 based on whether accepted locally or not
-        }
-
-        Task<Attestation> IBeaconNodeOApiController.AttestationAsync(byte[] validator_pubkey, int poc_bit, ulong slot, ulong shard)
+        // /// <summary>Publish a signed block.</summary>
```

Comment:
shall we remove the commented out code or is there a plan to use it?

Reply:
I do need to put back in the API, so that we can separate out validator & beacon node (there is a task/work item logged fro it); not sure which code will be reused, but we will need something.